cirq/google/engine/engine.py

```diff
@@ -76,29 +62,35 @@ def _make_random_id(prefix: str, length: int = 16):
 
 
 @value.value_equality
-class JobConfig:
-    """Configuration for a job to run on the Quantum Engine API.
-
-    An instance of a program that has been scheduled on the Quantum Engine is
-    called a Job. This object contains the configuration for a job.
-    """
+class EngineContext:
+    """Context for running against the Quantum Engine API."""
 
-    def __init__(self, job_id: Optional[str] = None) -> None:
+    def __init__(self,
+                 proto_version: Optional[ProtoVersion] = ProtoVersion.V2,
+                 service_args: Optional[Dict] = None,
+                 verbose: Optional[bool] = True,
+                 client: 'Optional[engine_client.EngineClient]' = None) -> None:
         """Configuration for a job that is run on Quantum Engine.
 
         Args:
```

Comment:
We should document common args, though maybe you want some service_args to be opaque?